packages/autocertifier-client/src/AutoCertifierClient.ts

```diff
@@ -0,0 +1,165 @@
+import { EventEmitter } from 'eventemitter3'
+import { IAutoCertifierService } from './proto/packages/autocertifier/protos/AutoCertifier.server'
+import { SessionIdRequest, SessionIdResponse } from './proto/packages/autocertifier/protos/AutoCertifier'
+import { ServerCallContext } from '@protobuf-ts/runtime-rpc'
+import { filePathToNodeFormat } from '@streamr/utils'
+import { RestClient } from './RestClient'
+import { CertifiedSubdomain } from './data/CertifiedSubdomain'
+import fs from 'fs'
+import path from 'path'
+import * as forge from 'node-forge'
+import { Logger } from '@streamr/utils'
+
+interface AutoCertifierClientEvents {
+    updatedSubdomain: (domain: CertifiedSubdomain) => void
+}
+
+const logger = new Logger(module)
+
+export const AUTOCERTIFIER_SERVICE_ID = 'AutoCertifier'
+
+export class AutoCertifierClient extends EventEmitter<AutoCertifierClientEvents> implements IAutoCertifierService {
+
+    private readonly ONE_DAY = 1000 * 60 * 60 * 24
+    private MAX_INT_32 = 2147483647
+    private updateTimeout?: NodeJS.Timeout
+    private readonly restClient: RestClient
+    private readonly subdomainPath: string
+    private readonly streamrWebSocketPort: number
+    private readonly ongoingSessions: Set<string> = new Set()
+
+    constructor(subdomainPath: string, streamrWebSocketPort: number, restApiUrl: string,
+        registerRpcMethod: (serviceId: string, rpcMethodName: string,
+            method: (request: SessionIdRequest, context: ServerCallContext) => Promise<SessionIdResponse>) => void) {
+
+        super()
+
+        this.restClient = new RestClient(restApiUrl)
+        this.subdomainPath = filePathToNodeFormat(subdomainPath)
+        this.streamrWebSocketPort = streamrWebSocketPort
+        registerRpcMethod(AUTOCERTIFIER_SERVICE_ID, 'getSessionId', this.getSessionId.bind(this))
+    }
+
+    public async start(): Promise<void> {
+        if (!fs.existsSync(this.subdomainPath)) {
+            await this.createCertificate()
+        } else {
+            await this.checkSubdomainValidity()
+        }
+    }
+
+    private async checkSubdomainValidity(): Promise<void> {
+        const sub = this.loadSubdomainFromDisk()
+
+        if (Date.now() >= sub.expiryTime - this.ONE_DAY) {
```

Comment:
This kind of naming convention is usually used:

```suggestion
        if (Date.now() >= sub.expirationTimestamp - this.ONE_DAY) {
```